### DISCUSSION TITLE
Suggestion: Support for Multiple API Keys for Automatic Token Limit Management

### DISCUSSION THREAD

**Discussion by gtillemann** [cline/cline#285] [Feature Requests]
After a couple of weeks of using the project and maximizing token usage, I found myself needing to create another account on Anthropic with a different email address to obtain an additional API key and continue leveraging Claude 3.5 Sonnet. My usage is so intensive that I reach the daily token limit before midday, and it is vital for me to replace the API key to continue writing code.

What is the idea?

To implement a feature that allows adding multiple API keys so that, upon reaching the maximum limit, it automatically switches to another key or balances the requests among them. Essentially, that's the idea.

The project is fantastic; many congratulations on this great work.

**Comment by alexxety** [comment]
I also use several keys, I wanted to offer this from the very beginning of the development of this application

**Reply by jitsmaster** [reply]
This is a great idea, since Claude pro plan always have rate limits

**Comment by gtillemann** [comment]
Well, it's still very useful and necessary. Hopefully, you can implement it.Best regards.-- Gino Tillemann +562 2760 5300Este correo electrónico y cualquier documento adjunto puede contener información confidencial de Masnet SpA, dirigida para el conocimiento y uso exclusivo de las personas o entidades arriba mencionadas. Si Usted no es el destinatario, se le informa que cualquier divulgación, distribución o copia de esta comunicación está estrictamente prohibida. Si usted ha recibido este correo electrónico por error, agradeceremos notificar respondiendo este mensaje y eliminando el original y cualquier copia o impresión de este.This communication contains information from Masnet SpA. that may be confidential. Except for personal use by the intended recipient, or as expressly authorized by the sender, any person who receives this information is prohibited from disclosing, copying, distributing, and/or using it. If you have received this communication in error, please immediately delete it and all copies, and promptly notify the sender. Nothing in this communication is intended to operate as an electronic signature under applicable law.Chile, Ley Nº 19628 de Protección de Datos Personales de 28 de Agosto de 1999 y sus modificaciones vigentes.    ---- On sáb., 14 sept. 2024 13:41:16 -0300  ***@***.***> wrote ----  
I also use several keys, I wanted to offer this from the very beginning of the development of this application

**Comment by Sarke** [comment]
I think balancing among the keys would make the caching less useful, right? I also assume it's against TOS, but that is what it is.

**Comment by gtillemann** [comment]
It could be as simple as switching to a new registered API key once the daily token limit is reached. There’s nothing against the terms of service; you're paying for each API key.



Gino









---- El mar., 24 sept. 2024 20:39:48 -0300, Peter Stalman ***@***.***> escribió ----





I think balancing among the keys would make the caching less useful, right? I also assume it's against TOS, but that is what it is.

**Comment by TomLucidor** [comment]
Add something extra for FOSS and token reduction too, and if open models aren't satisfactory sometimes, at least less money was wasted most of the time https://github.com/lm-sys/RouteLLM https://github.com/aurelio-labs/semantic-router

**Comment by burc330012** [comment]
This is what I'm always doing manually for claude every day, I hope someone can implement this api key rotation for models, this would be very usefull

**Comment by gtillemann** [comment]
It's better to use Roo Code; with this extension, I can create an endless number of models.







Gino







---- El vie., 24 ene. 2025 15:38:43 -0300, Arnold Wang ***@***.***> escribió ----





This is a great idea, since Claude pro plan always have rate limits
